app/js/docker/dockerService.js

```diff
@@ -66,6 +66,11 @@ function dockerService($http, actions, flux, alertService, configService) {
             }
         };
 
+        if (action === 'pullImage') {
+            // hack for pulling images
+            config.params = data;
+        }
+
```

Comment:
Yeah I wasn't thinking about POST requests with query parameters when I wrote this :) What if we did something like:

```
function d(method, url, action, host, cid, data) {
    return _.partial(...);
}
```

where data could be an object:

```
{
    'query': { <query parameter map> },
    'post': <post data>
}
```

and then the meat of the docker function above could just apply both (if defined)?